src/bundle/Controller/LocationController.php

```diff
@@ -83,6 +85,26 @@ public function actionsAction(
         return $this->reloadTab('locations', $content->id, $redirectLocationId);
     }
 
+    public function swapLocationAction(
+        Content $content,
+        Location $location,
+        Request $request
+    ) {
+        $swapLocationsForm = $this->formFactory->createLocationsContentSwapForm();
+        $swapLocationsForm->handleRequest($request);
+
+        if ($swapLocationsForm->isValid()) {
+            $newLocationId = $swapLocationsForm->get('new_location_id')->getData();
+            $this->uiLocationService->swapLocations($location, $newLocationId);
+
+            return $this->resetLocation($content->id);
+        }
+
+        $redirectLocationId = $request->query->get('redirectLocationId', $content->contentInfo->mainLocationId);
+
+        return $this->reloadTab('locations', $content->id, $redirectLocationId);
```

Comment:
I don't think `reloadTab()` will work here.

We are viewing `Location A`, referring to `Content X`, and we are swapping it with `Location B`, that refers to `Content Y`. After the swap, `Location A` refers to `Content Y`, and `Location B` refers to `Content X`.

If we only reload the tab, we are still viewing `Location A`, and only the Locations tab is reloaded. Most, if not all, of the other tabs are incorrect, since they still refer to `Content X`. We should call `resetLocation()`, either with the originally viewed location, or the one we swapped it for.

Also, why do we need a `redirectLocationId` ? We know about both of the involved locations, and shouldn't need that.